### ISSUE TITLE
[Bug]: Experimenter form edge case

### ISSUE THREAD

**Issue by CodyCBakerPhD** [NeurodataWithoutBorders/nwbinspector#409]
### What happened?

Discovered during the first official alpha test of NWB GUIDE

```python
from nwbinspector import check_experimenter_form
from pynwb.testing.mock.file import mock_NWBFile

nwbfile = mock_NWBFile(experimenter="Doe, John.")
check_experimenter_form(nwbfile=nwbfile)
> None
```

But this should have been simply `Doe, John`

Need to check why the regex doesn't flag this; I had thought it would look only for a space before a letter then the period which is allowed for middle initials

### Operating System

None

### Python Version

None

### Were you streaming with ROS3?

None

### Package Versions

<details> Copy and paste your output here </details>

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/neurodatawithoutborders/nwbinspector/issues)?
- [X] To the best of your ability, have you ensured this is a bug within the code that checks the NWBFile, rather than a bug in the NWBFile reader (e.g., [PyNWB](https://github.com/NeurodataWithoutBorders/pynwb) or [MatNWB](https://github.com/NeurodataWithoutBorders/matnwb))?
